`develop` branch broken on oneAPI

(initially reported by @AuroraPerego)

Hi,
looks like the current HEAD of the `develop` branch is broken for the SYCL/oneAPI GPU target:

```
84% tests passed, 7 tests failed out of 43

Total Test time (real) =  70.06 sec

The following tests FAILED:
          8 - parallelLoopPatterns (Subprocess aborted)
         10 - randomCells2D (Failed)
         17 - matMulTest (Failed)
         32 - memBufTest (Not Run)
         33 - bufSlicingTest (Subprocess aborted)
         36 - memViewTest (Subprocess aborted)
         42 - warpTest (Failed)
```

This is one an Intel Data Center GPU Max 1100 (Ponte Vecchio) with oneAPI DPC++/C++ Compiler 2023.2.0 (or 2023.2.1, there seems to be some confusion with minor versions):
```
$ git log --oneline -n1
59002235d6d (HEAD -> develop, origin/develop, origin/HEAD) Fix unsigned integer conversion

$ mkdir -p build/sycl_gpu

$ cd build/sycl_gpu

$ CXXFLAGS="-g -O2" cmake \
  -G 'Unix Makefiles' \
  -DCMAKE_CXX_COMPILER=/opt/intel/oneapi/compiler/latest/linux/bin/icpx \
  -DoneDPL_ROOT=/opt/intel/oneapi/dpl/latest \
  -DoneDPL_DIR=/opt/intel/oneapi/dpl/latest/lib/cmake/oneDPL \
  -DTBB_ROOT=/opt/intel/oneapi/tbb/latest \
  -DTBB_DIR=/opt/intel/oneapi/tbb/latest/lib/cmake/tbb \
  -DBOOST_ROOT=~/local/boost/ \
  --log-level=VERBOSE \
  -Dalpaka_DEBUG=2 \
  -Dalpaka_BUILD_EXAMPLES=ON \
  -Dalpaka_CHECK_HEADERS=ON \
  -DBUILD_TESTING=ON \
  -DCMAKE_BUILD_TYPE=Debug \
  -DCMAKE_VERBOSE_MAKEFILE=ON \
  -Dalpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE=ON \
  -Dalpaka_ACC_SYCL_ENABLE=ON \
  -Dalpaka_SYCL_PLATFORM_ONEAPI=ON \
  -Dalpaka_SYCL_ONEAPI_CPU=OFF \
  -Dalpaka_SYCL_ONEAPI_GPU=ON \
  -Dalpaka_SYCL_ONEAPI_GPU_DEVICES='intel_gpu_pvc' \
  -Dalpaka_DISABLE_VENDOR_RNG=ON \
  -L \
  ../../

$ make -j8 -k

$ make test
```

### `memBufTest`

`zeroDimBufferTest` fails to build with
```
$ make
...
[ 80%] Building CXX object test/integ/zeroDimBuffer/CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o
cd /home/u106132/alpaka/build/sycl_gpu/test/integ/zeroDimBuffer && /opt/intel/oneapi/compiler/latest/linux/bin/icpx -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED -DALPAKA_ACC_SYCL_ENABLED -DALPAKA_BLOCK_SHARED_DYN_MEMBER_ALLOC_KIB=47 -DALPAKA_DEBUG=2 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_OFFLOAD_MAX_BLOCK_SIZE="" -DALPAKA_SYCL_ONEAPI_GPU -DALPAKA_SYCL_TARGET_GPU -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -I/home/u106132/alpaka/include -isystem /home/u106132/alpaka/thirdParty/catch2/src/catch2/.. -isystem /home/u106132/alpaka/build/sycl_gpu/thirdParty/catch2/generated-includes -isystem /home/u106132/local/boost/include -g -O2 -g -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-global-constructors -Wno-padded -Wno-extra-semi-stmt -ffp-model=precise -Wno-disabled-macro-expansion -Wno-unsafe-buffer-usage -O0 -fsycl -sycl-std=2020 -fsycl-targets=intel_gpu_pvc -fsycl-unnamed-lambda -MD -MT test/integ/zeroDimBuffer/CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o -MF CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o.d -o CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o -c /home/u106132/alpaka/test/integ/zeroDimBuffer/src/zeroDimBuffer.cpp
In file included from /home/u106132/alpaka/test/integ/zeroDimBuffer/src/zeroDimBuffer.cpp:9:
In file included from /home/u106132/alpaka/include/alpaka/alpaka.hpp:13:
In file included from /home/u106132/alpaka/include/alpaka/acc/AccCpuOmp2Blocks.hpp:24:
In file included from /home/u106132/alpaka/include/alpaka/workdiv/WorkDivMembers.hpp:8:
/home/u106132/alpaka/include/alpaka/extent/Traits.hpp:92:5: error: static assertion failed due to requirement 'integral_constant<unsigned long, 0>::value >= 1'
    static_assert(Dim<TExtent>::value >= 1);
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~
```

### `parallelLoopPatterns`
`parallelLoopPatterns` fails at runtime with an assertion:
```
$ ./example/parallelLoopPatterns/parallelLoopPatterns
...
parallelLoopPatterns: /home/u106132/alpaka/include/alpaka/mem/buf/sycl/Set.hpp:53: alpaka::detail::TaskSetSyclBase<std::integral_constant<unsigned long, 1>, alpaka::BufGenericSycl<float, std::integral_constant<unsigned long, 1>, unsigned int, alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, alpaka::Vec<std::integral_constant<unsigned long, 1>, unsigned int>>::TaskSetSyclBase(TViewFwd &&, const std::uint8_t &, const TExtent &) [TDim = std::integral_constant<unsigned long, 1>, TView = alpaka::BufGenericSycl<float, std::integral_constant<unsigned long, 1>, unsigned int, alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, TExtent = alpaka::Vec<std::integral_constant<unsigned long, 1>, unsigned int>, TViewFwd = alpaka::BufGenericSycl<float, std::integral_constant<unsigned long, 1>, unsigned int, alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>> &]: Assertion `m_extentWidthBytes <= m_dstPitchBytes[TDim::value - 1u]' failed.
Aborted (core dumped)
```

### `bufSlicingTest`
`bufSlicingTest` fails at runtime with an assertion:
```
$ ./test/unit/mem/copy/bufSlicingTest
...
bufSlicingTest: /home/u106132/alpaka/include/alpaka/mem/buf/sycl/Set.hpp:53: alpaka::detail::TaskSetSyclBase<std::integral_constant<unsigned long, 1>, alpaka::ViewSubView<alpaka::DevGenericSycl<alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, int, std::integral_constant<unsigned long, 1>, long>, alpaka::Vec<std::integral_constant<unsigned long, 1>, long>>::TaskSetSyclBase(TViewFwd &&, const std::uint8_t &, const TExtent &) [TDim = std::integral_constant<unsigned long, 1>, TView = alpaka::ViewSubView<alpaka::DevGenericSycl<alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, int, std::integral_constant<unsigned long, 1>, long>, TExtent = alpaka::Vec<std::integral_constant<unsigned long, 1>, long>, TViewFwd = alpaka::ViewSubView<alpaka::DevGenericSycl<alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, int, std::integral_constant<unsigned long, 1>, long> &]: Assertion `m_extentWidthBytes <= m_dstPitchBytes[TDim::value - 1u]' failed.
-------------------------------------------------------------------------------
memBufSlicingMemsetTest - TestAccWithDataTypes - 1
-------------------------------------------------------------------------------
/home/u106132/alpaka/test/unit/mem/copy/src/BufSlicing.cpp:170
...............................................................................

/home/u106132/alpaka/test/unit/mem/copy/src/BufSlicing.cpp:170: FAILED:
due to a fatal error condition:
  SIGABRT - Abort (abnormal termination) signal

===============================================================================
test cases:   74 |   49 passed | 25 failed
assertions: 3105 | 3080 passed | 25 failed

Aborted (core dumped)
```

### `memViewTest`
`memViewTest` fails at runtime with the same assertion:
```
$ ./test/unit/mem/view/memViewTest
...
memViewTest: /home/u106132/alpaka/include/alpaka/mem/buf/sycl/Set.hpp:53: alpaka::detail::TaskSetSyclBase<std::integral_constant<unsigned long, 1>, alpaka::ViewPlainPtr<alpaka::DevGenericSycl<alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, float, std::integral_constant<unsigned long, 1>, long>, alpaka::Vec<std::integral_constant<unsigned long, 1>, long>>::TaskSetSyclBase(TViewFwd &&, const std::uint8_t &, const TExtent &) [TDim = std::integral_constant<unsigned long, 1>, TView = alpaka::ViewPlainPtr<alpaka::DevGenericSycl<alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, float, std::integral_constant<unsigned long, 1>, long>, TExtent = alpaka::Vec<std::integral_constant<unsigned long, 1>, long>, TViewFwd = alpaka::ViewPlainPtr<alpaka::DevGenericSycl<alpaka::PlatformGenericSycl<alpaka::detail::IntelGpuSelector>>, float, std::integral_constant<unsigned long, 1>, long> &]: Assertion `m_extentWidthBytes <= m_dstPitchBytes[TDim::value - 1u]' failed.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
memViewTest is a Catch2 v3.3.2 host application.
Run with -? for options

-------------------------------------------------------------------------------
viewPlainPtrTest - alpaka::test::TestAccs - 1
-------------------------------------------------------------------------------
/home/u106132/alpaka/test/unit/mem/view/src/ViewPlainPtrTest.cpp:97
...............................................................................

/home/u106132/alpaka/test/unit/mem/view/src/ViewPlainPtrTest.cpp:97: FAILED:
  {Unknown expression after the reported line}
due to a fatal error condition:
  SIGABRT - Abort (abnormal termination) signal

===============================================================================
test cases:   26 |   25 passed | 1 failed
assertions: 1494 | 1493 passed | 1 failed

Aborted (core dumped)
```

### `randomCells2D`
`randomCells2D` fails at runtime:
```
$ ./example/randomCells2D/randomCells2D
...
Number of cells: 26797
Number of calculations per cell: 256
Total number of calculations: 6860032
Mean value A: 0.00628226 (should converge to 0.5)
Mean value B: 0.00628699 (should converge to 0.5)
Maximum error expected at 6860032 calculations should be around 0.0001909
Convergence test failed!
...

$ echo $?
1
```

### `matMulTest`
`matMulTest` fails at runtime:
```
$ ./test/integ/matMul/matMulTest
...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
matMulTest is a Catch2 v3.3.2 host application.
Run with -? for options

-------------------------------------------------------------------------------
matMul - TestAccs - 1
-------------------------------------------------------------------------------
/home/u106132/alpaka/test/integ/matMul/src/matMul.cpp:153
...............................................................................

/home/u106132/alpaka/test/integ/matMul/src/matMul.cpp:306: FAILED:
  REQUIRE( resultCorrect )
with expansion:
  false

[+] ~BufCpuImpl
[-] ~BufCpuImpl
===============================================================================
test cases: 2 | 1 passed | 1 failed
assertions: 8 | 7 passed | 1 failed
```

### `warpTest`
`warpTest` fails at runtime:
```
$ ./test/unit/warp/warpTest
...
-------------------------------------------------------------------------------
shfl - alpaka::test::TestAccs - 23
-------------------------------------------------------------------------------
/home/u106132/alpaka/test/unit/warp/src/Shfl.cpp:97
...............................................................................

/home/u106132/alpaka/test/unit/warp/src/Shfl.cpp:134: FAILED:
  REQUIRE( fixture(ShflMultipleThreadWarpTestKernel<16>{}) )
with expansion:
  false

===============================================================================
test cases: 144 |  84 passed |  60 failed
assertions: 732 | 108 passed | 624 failed
```

> memBufTest
>  zeroDimBufferTest fails to build with
>  ```
>  $ make
>  ...
>  [ 80%] Building CXX object test/integ/zeroDimBuffer/CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o
> cd /home/u106132/alpaka/build/sycl_gpu/test/integ/zeroDimBuffer && /opt/intel/oneapi/compiler/latest/linux/bin/icpx -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED -DALPAKA_ACC_SYCL_ENABLED -DALPAKA_BLOCK_SHARED_DYN_MEMBER_ALLOC_KIB=47 -DALPAKA_DEBUG=2 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_OFFLOAD_MAX_BLOCK_SIZE="" -DALPAKA_SYCL_ONEAPI_GPU -DALPAKA_SYCL_TARGET_GPU -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -I/home/u106132/alpaka/include -isystem /home/u106132/alpaka/thirdParty/catch2/src/catch2/.. -isystem /home/u106132/alpaka/build/sycl_gpu/thirdParty/catch2/generated-includes -isystem /home/u106132/local/boost/include -g -O2 -g -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-global-constructors -Wno-padded -Wno-extra-semi-stmt -ffp-model=precise -Wno-disabled-macro-expansion -Wno-unsafe-buffer-usage -O0 -fsycl -sycl-std=2020 -fsycl-targets=intel_gpu_pvc -fsycl-unnamed-lambda -MD -MT test/integ/zeroDimBuffer/CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o -MF CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o.d -o CMakeFiles/zeroDimBufferTest.dir/src/zeroDimBuffer.cpp.o -c /home/u106132/alpaka/test/integ/zeroDimBuffer/src/zeroDimBuffer.cpp
>In file included from /home/u106132/alpaka/test/integ/zeroDimBuffer/src/zeroDimBuffer.cpp:9:
>In file included from /home/u106132/alpaka/include/alpaka/alpaka.hpp:13:
>In file included from /home/u106132/alpaka/include/alpaka/acc/AccCpuOmp2Blocks.hpp:24:
>In file included from /home/u106132/alpaka/include/alpaka/workdiv/WorkDivMembers.hpp:8:
>/home/u106132/alpaka/include/alpaka/extent/Traits.hpp:92:5: error: static assertion failed due to requirement >'integral_constant<unsigned long, 0>::value >= 1'
>    static_assert(Dim<TExtent>::value >= 1);
>    ^             ~~~~~~~~~~~~~~~~~~~~~~~~
>  ```

I do not see it in the tests, but I have seen pixeltrack failing in the same way after [this](https://github.com/alpaka-group/alpaka/pull/2080/commits/b195bc4361c0777558e2b397e616405fa9b3729c) commit

--- a/include/alpaka/mem/buf/cpu/Copy.hpp
+++ b/include/alpaka/mem/buf/cpu/Copy.hpp
@@ -106,8 +106,10 @@ namespace alpaka
                 // [z, y, x] -> [z, y] because all elements with the innermost x dimension are handled within one
                 // iteration.
                 Vec<DimMin1, ExtentSize> const extentWithoutInnermost = subVecBegin<DimMin1>(this->m_extent);
-                Vec<DimMin1, DstSize> const dstPitchBytesWithoutOutmost = subVecBegin<DimMin1>(this->m_dstPitchBytes);
-                Vec<DimMin1, SrcSize> const srcPitchBytesWithoutOutmost = subVecBegin<DimMin1>(this->m_srcPitchBytes);
+                Vec<DimMin1, DstSize> const dstPitchBytesWithoutInnermost
+                    = subVecBegin<DimMin1>(this->m_dstPitchBytes);
+                Vec<DimMin1, SrcSize> const srcPitchBytesWithoutInnermost
+                    = subVecBegin<DimMin1>(this->m_srcPitchBytes);
 
                 if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
                 {
@@ -116,8 +118,8 @@ namespace alpaka
                         [&](Vec<DimMin1, ExtentSize> const& idx)
                         {
                             std::memcpy(
-                                this->m_dstMemNative + (castVec<DstSize>(idx) * dstPitchBytesWithoutOutmost).sum(),
-                                this->m_srcMemNative + (castVec<SrcSize>(idx) * srcPitchBytesWithoutOutmost).sum(),
+                                this->m_dstMemNative + (castVec<DstSize>(idx) * dstPitchBytesWithoutInnermost).sum(),
+                                this->m_srcMemNative + (castVec<SrcSize>(idx) * srcPitchBytesWithoutInnermost).sum(),
                                 static_cast<std::size_t>(this->m_extentWidthBytes));
                         });
                 }

--- a/include/alpaka/mem/buf/cpu/Set.hpp
+++ b/include/alpaka/mem/buf/cpu/Set.hpp
@@ -117,7 +117,7 @@ namespace alpaka
                 if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
                 {
                     std::memset(
-                        reinterpret_cast<void*>(this->m_dstMemNative),
+                        this->m_dstMemNative,
                         this->m_byte,
                         static_cast<std::size_t>(this->m_extentWidthBytes));
                 }
@@ -159,7 +159,7 @@ namespace alpaka
 #if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
                 printDebug();
 #endif
-                std::memset(reinterpret_cast<void*>(m_dstMemNative), m_byte, sizeof(Elem));
+                std::memset(m_dstMemNative, m_byte, sizeof(Elem));
             }
 
             std::uint8_t const m_byte;

--- a/include/alpaka/mem/buf/sycl/Copy.hpp
+++ b/include/alpaka/mem/buf/sycl/Copy.hpp
@@ -41,7 +41,7 @@ namespace alpaka::detail
         template<typename TViewFwd>
         TaskCopySyclBase(TViewFwd&& viewDst, TViewSrc const& viewSrc, TExtent const& extent)
             : m_extent(getExtents(extent))
-            , m_extentWidthBytes(m_extent[TDim::value - 1u] * static_cast<ExtentSize>(sizeof(Elem)))
+            , m_extentWidthBytes(m_extent.back() * static_cast<ExtentSize>(sizeof(Elem)))
 #    if(!defined(NDEBUG)) || (ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL)
             , m_dstExtent(getExtents(viewDst))
             , m_srcExtent(getExtents(viewSrc))
@@ -53,8 +53,8 @@ namespace alpaka::detail
         {
             if constexpr(TDim::value > 0)
             {
-                ALPAKA_ASSERT((castVec<DstSize>(m_extent) <= m_dstExtent).foldrAll(std::logical_or<bool>()));
-                ALPAKA_ASSERT((castVec<SrcSize>(m_extent) <= m_srcExtent).foldrAll(std::logical_or<bool>()));
+                ALPAKA_ASSERT((castVec<DstSize>(m_extent) <= m_dstExtent).all());
+                ALPAKA_ASSERT((castVec<SrcSize>(m_extent) <= m_srcExtent).all());
             }
         }
 
@@ -103,9 +103,8 @@ namespace alpaka::detail
             // [z, y, x] -> [z, y] because all elements with the innermost x dimension are handled within one
             // iteration.
             Vec<DimMin1, ExtentSize> const extentWithoutInnermost(subVecBegin<DimMin1>(this->m_extent));
-            // [z, y, x] -> [y, x] because the z pitch (the full size of the buffer) is not required.
-            Vec<DimMin1, DstSize> const dstPitchBytesWithoutOutmost(subVecEnd<DimMin1>(this->m_dstPitchBytes));
-            Vec<DimMin1, SrcSize> const srcPitchBytesWithoutOutmost(subVecEnd<DimMin1>(this->m_srcPitchBytes));
+            Vec<DimMin1, DstSize> const dstPitchBytesWithoutInnermost(subVecBegin<DimMin1>(this->m_dstPitchBytes));
+            Vec<DimMin1, SrcSize> const srcPitchBytesWithoutInnermost(subVecBegin<DimMin1>(this->m_srcPitchBytes));
 
             // Record an event for each memcpy call
             std::vector<sycl::event> events;
@@ -118,14 +117,8 @@ namespace alpaka::detail
                     [&](Vec<DimMin1, ExtentSize> const& idx)
                     {
                         events.push_back(queue.memcpy(
-                            reinterpret_cast<void*>(
-                                this->m_dstMemNative
-                                + (castVec<DstSize>(idx) * dstPitchBytesWithoutOutmost)
-                                      .foldrAll(std::plus<DstSize>())),
-                            reinterpret_cast<void const*>(
-                                this->m_srcMemNative
-                                + (castVec<SrcSize>(idx) * srcPitchBytesWithoutOutmost)
-                                      .foldrAll(std::plus<SrcSize>())),
+                            this->m_dstMemNative + (castVec<DstSize>(idx) * dstPitchBytesWithoutInnermost).sum(),
+                            this->m_srcMemNative + (castVec<SrcSize>(idx) * srcPitchBytesWithoutInnermost).sum(),
                             static_cast<std::size_t>(this->m_extentWidthBytes),
                             requirements));
                     });
@@ -154,8 +147,8 @@ namespace alpaka::detail
             if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
             {
                 return queue.memcpy(
-                    reinterpret_cast<void*>(this->m_dstMemNative),
-                    reinterpret_cast<void const*>(this->m_srcMemNative),
+                    this->m_dstMemNative,
+                    this->m_srcMemNative,
                     sizeof(Elem) * static_cast<std::size_t>(this->m_extent.prod()),
                     requirements);
             }

--- a/include/alpaka/mem/buf/sycl/Set.hpp
+++ b/include/alpaka/mem/buf/sycl/Set.hpp
@@ -40,7 +40,7 @@ namespace alpaka
             TaskSetSyclBase(TViewFwd&& view, std::uint8_t const& byte, TExtent const& extent)
                 : m_byte(byte)
                 , m_extent(getExtents(extent))
-                , m_extentWidthBytes(m_extent[TDim::value - 1u] * static_cast<ExtentSize>(sizeof(Elem)))
+                , m_extentWidthBytes(m_extent.back() * static_cast<ExtentSize>(sizeof(Elem)))
 #    if(!defined(NDEBUG)) || (ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL)
                 , m_dstExtent(getExtents(view))
 #    endif
@@ -49,8 +49,9 @@ namespace alpaka
                 , m_dstMemNative(reinterpret_cast<std::uint8_t*>(getPtrNative(view)))
 
             {
-                ALPAKA_ASSERT((castVec<DstSize>(m_extent) <= m_dstExtent).foldrAll(std::logical_or<bool>()));
-                ALPAKA_ASSERT(m_extentWidthBytes <= m_dstPitchBytes[TDim::value - 1u]);
+                ALPAKA_ASSERT((castVec<DstSize>(m_extent) <= m_dstExtent).all());
+                if constexpr(TDim::value > 1)
+                    ALPAKA_ASSERT(m_extentWidthBytes <= m_dstPitchBytes[TDim::value - 2]);
             }
 
 #    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
@@ -93,8 +94,7 @@ namespace alpaka
                 // [z, y, x] -> [z, y] because all elements with the innermost x dimension are handled within one
                 // iteration.
                 Vec<DimMin1, ExtentSize> const extentWithoutInnermost(subVecBegin<DimMin1>(this->m_extent));
-                // [z, y, x] -> [y, x] because the z pitch (the full idx of the buffer) is not required.
-                Vec<DimMin1, DstSize> const dstPitchBytesWithoutOutmost(subVecEnd<DimMin1>(this->m_dstPitchBytes));
+                Vec<DimMin1, DstSize> const dstPitchBytesWithoutInnermost(subVecBegin<DimMin1>(this->m_dstPitchBytes));
 
                 // Record an event for each memcpy call
                 std::vector<sycl::event> events;
@@ -107,10 +107,7 @@ namespace alpaka
                         [&](Vec<DimMin1, ExtentSize> const& idx)
                         {
                             events.push_back(queue.memset(
-                                reinterpret_cast<void*>(
-                                    this->m_dstMemNative
-                                    + (castVec<DstSize>(idx) * dstPitchBytesWithoutOutmost)
-                                          .foldrAll(std::plus<DstSize>())),
+                                this->m_dstMemNative + (castVec<DstSize>(idx) * dstPitchBytesWithoutInnermost).sum(),
                                 this->m_byte,
                                 static_cast<std::size_t>(this->m_extentWidthBytes),
                                 requirements));

